File: openfinalfinal/u202120418/u202120418/src/main/java/org/hign/platform/u202120418/assessment/interfaces/rest/transform/MentalStateExamResourceFromEntityAssembler.java
package org.hign.platform.u202120418.assessment.interfaces.rest.transform;/**
 * @author Fiorella Jarama Peñaloza
 * @version 1.0
 */

import org.hign.platform.u202120418.assessment.domain.model.aggregates.MentalStateExam;
import org.hign.platform.u202120418.assessment.interfaces.rest.resources.MentalStateExamResource;

/**
 * Class that transforms a MentalStateExam entity into a MentalStateExamResource.
 *
 * @author Fiorella Jarama Peñaloza
 */
public class MentalStateExamResourceFromEntityAssembler {
    public static MentalStateExamResource toResourceFromEntity(MentalStateExam entity) {
        return new MentalStateExamResource(
                entity.getId(),
                entity.getPatientId(),
                entity.getExamDate(),
                entity.getOrientationScore(),
                entity.getAttentionAndCalculationScore(),
                entity.getRegistrationScore(),
                entity.getRecallScore(),
                entity.getLanguageScore(),
                entity.getExaminer()
        );
    }
}

File: web/HIGN/HIGN/Assessment/Interfaces/REST/Transform/MentalStateExamResourceFromEntityAssembler.cs
using HIGN.Assessment.Domain.Model.Aggregates;
using HIGN.Assessment.Interfaces.REST.Resources;

namespace HIGN.Assessment.Interfaces.REST.Transform;

public class MentalStateExamResourceFromEntityAssembler
{
    public static MentalStateExamResource ToResourceFromEntity(MentalStateExam mentalStateExam)
    {
        return new MentalStateExamResource(mentalStateExam.Id, mentalStateExam.PatientId,
            mentalStateExam.NationalProvideIdentifier,
            mentalStateExam.ExamDate, mentalStateExam.OrientationScore, mentalStateExam.RegistrationScore,
            mentalStateExam.AttentionAndCalculationScore, mentalStateExam.RecallScore,
            mentalStateExam.LanguageScore, mentalStateExam.ExaminerId);
    }
}